chaste_codegen/chaste_model.py

```diff
@@ -827,3 +416,7 @@ def generate_chaste_code(self):
         # Generate cpp for model
         template = cg.load_template(self._cpp_template)
         self.generated_cpp = template.render(self._vars_for_template)
+
+    def __exit__(self, type, value, traceback):
```

Comment:
Does this get overridden in subclasses? Because it looks pointless here!

Reply:
yes but only in `CvodeWithDataClampModel` the reason it's there is that I added the data clamp to the cellmlmanip model, so have to remove it again for the model object to be reusable. I added this to the base class more as a signature and reminder in case we add more types in future really.

Reply:
I would add to the docstring for `__enter__` and `__exit__` that they are defined simply here but may be overridden in sub-classes if state needs to be reset for subsequent code generation.

Reply:
Ok I've also made a note of that these are required wo be able to use them in context.
The way the cleanup for the data clamp model works ia that the model is used  in context in the console script. I tried doing it in a destructor, but I couldn't be sure of exactly when this would be called and it's quite important it happens before subsequent model generations

```
with translator_class(model, outfile_base, header_ext=ext[0], **vars(args)) as chaste_model:
                chaste_model.generate_chaste_code()
```